base/essentials.jl

```diff
@@ -54,12 +54,12 @@ end
     @nospecialize
 
 Applied to a function argument name, hints to the compiler that the method
-should not be specialized for different types of that argument,
-but instead to use precisely the declared type for each argument.
-This is only a hint for avoiding excess code generation.
-Can be applied to an argument within a formal argument list,
+implementation should not be specialized for different types of that argument,
+but instead use the declared type for that argument.
+It can be applied to an argument within a formal argument list,
 or in the function body.
-When applied to an argument, the macro must wrap the entire argument expression.
+When applied to an argument, the macro must wrap the entire argument expression, e.g.,
+`@nospecialize(x::Real)` or `@nospecialize(i::Integer...)` rather than wrapping just the argument name.
```

Comment:
This is not necessarily the case if the function only takes one argument:

```julia
julia> function h(@nospecialize x::Int)
       x
       end
h (generic function with 1 method)

julia> function f(@nospecialize x)
       x
       end
f (generic function with 1 method)
```
Not sure whether it's worth it to elaborate on this small technicality here, but maybe we could give an example below that shows this way, since it is a bit cleaner imo

Reply:
They are general syntactical variants of syntax for invoking macros and I don't think we necessarily want to discuss it here again (as it's described [in the manual](https://docs.julialang.org/en/v1/manual/metaprogramming/#Macro-invocation)).